ulauncher/modes/extensions/extension_mode.py

```diff
@@ -155,3 +155,104 @@ def trigger_event(self, event: dict[str, Any]) -> None:
+
+        ext_id = payload.get("ext_id")
+        path = payload.get("path")
+        with_debugger = bool(payload.get("with_debugger"))
+        assert ext_id, "preview_ext called without ext_id"
+        assert path, "preview_ext called without path"
+
+        logger.info(
+            "[preview] Received preview request for ext_id=%s path=%s debugger=%s",
+            ext_id,
+            path,
+            with_debugger,
+        )
+
+        try:
+            existing_controller = extension_registry.get(ext_id)
+        except ExtensionNotFoundError:
+            existing_controller = None
+        if existing_controller and existing_controller.is_running:
+            logger.info(
+                "[preview] Extension '%s' is currently running; stopping it before launching preview version",
+                ext_id,
+            )
+            self.run_ext_batch_job([ext_id], ["stop"], done_msg=f"[preview] Extension '{ext_id}' stopped")
+
+        preview_ext_id = f"{ext_id}.preview"
+        controller = extension_registry.load(preview_ext_id, path)
+        controller.is_preview = True
+
+        # install python dependencies from requirements.txt
+        from ulauncher.modes.extensions.extension_dependencies import ExtensionDependencies
+
+        deps = ExtensionDependencies(controller.id, controller.path)
+        deps.install()
+
+        # Run start_detached instead of start to avoid blocking the main thread
+        controller.start_detached(with_debugger=with_debugger)
+
+        logger.info("[preview] Preview extension '%s' started successfully", preview_ext_id)
+
+    @events.on
+    def stop_preview(self, payload: dict[str, Any] | None = None) -> None:
+        """Handle stopping a preview extension and restoring the previous version if any.
+
+        Expected payload example:
+            {
+              "preview_ext_id": "my-extension.preview",
+              "original_ext_id": "my-extension"
+            }
+        """
+        if not payload or not isinstance(payload, dict):
+            logger.error("stop_preview called without valid payload: %s", payload)
+            return
+
+        preview_ext_id = payload.get("preview_ext_id")
+        original_ext_id = payload.get("original_ext_id")
+
+        if not preview_ext_id or not original_ext_id:
+            logger.error("stop_preview called without required fields: %s", payload)
+            return
+
+        logger.info(
+            "[preview] Received stop preview request for preview_ext_id=%s, original_ext_id=%s",
+            preview_ext_id,
+            original_ext_id,
+        )
+
+        # Stop the preview extension
+        stop_msg = f"[preview] Preview extension '{preview_ext_id}' stopped"
+        self.run_ext_batch_job([preview_ext_id], ["stop"], done_msg=stop_msg)
+
+        # Try to restart the original extension
+        try:
+            original_controller = extension_registry.get(original_ext_id)
+        except ExtensionNotFoundError:
```

Comment:
With the changes made to the get method, this should be `except KeyError` right?